### ISSUE TITLE
Empty .gitmodules files cause a non-zero exit code

### ISSUE THREAD

**Issue by smirkybg** [Enteee/git-submodule-url-rewrite#1]
Hello,

I just stumbled accross a weird bug.
I have this project with recursive submodules. One of the submodules has an empty `.gitmodules` file. After running a recursive url rewrite, sed exits with a non-zero exit code and fails the whole rewrite script. I found that using --verbose.
Below, you will find an example log with obfuscated paths and names:
```
Entering 'path/to/submoduleX'
+ url_rewrite
+ set -euo pipefail
+ '[' true = true ']'
+ set -x
+ local toplevel
++ git rev-parse --show-toplevel
+ toplevel=/var/jenkins/workspace/blablabla/path/to/submoduleX
+ local gitmodules
+ gitmodules=/var/jenkins/workspace/blablabla/path/to/submoduleX/.gitmodules
+ cd /var/jenkins/workspace/blablabla/path/to/submoduleX
+ '[' '!' -r /var/jenkins/workspace/blablabla/path/to/submoduleX/.gitmodules ']'
+ sed -nre 's/^\[submodule \"(.*)\"]$/\1\x0/p' /var/jenkins/workspace/blablabla/path/to/submoduleX/.gitmodules
+ IFS=
+ read -rd '' submodule
+ xargs -0 -n1 printf '%b\0'
+ tr -d '\n'
+ submodule_url_rewrite ''
+ local name
+ name=
+ shift
+ local toplevel
++ git rev-parse --show-toplevel
+ toplevel=/var/jenkins/workspace/blablabla/path/to/submoduleX
+ local old_url
++ git config --file=/var/jenkins/workspace/blablabla/path/to/submoduleX/.gitmodules submodule..url
+ old_url=
Stopping at 'path/to/submoduleX'; script returned non-zero status.
```

**Comment by smirkybg** [comment]
Actually, it's not sed that fails. Anyway, I patched it like this:

`sed -i '115s/.*/    if [ ! -r "${gitmodules}" ] || [ ! -s "${gitmodules}" ]; then/' git-submodule-url-rewrite`

You may decide if this is indeed a legit case and bugfix.

**Comment by Enteee** [comment]
fixed with #2 . Thank you.
